cvat/apps/engine/task.py

```diff
@@ -272,13 +272,17 @@ def rq_handler(job, exc_type, exc_value, traceback):
 ############################# Internal implementation for server API
 
 class _FrameExtractor:
-    def __init__(self, source_path, compress_quality, flip_flag=False):
+    def __init__(self, source_path, compress_quality, flip_flag=False, frame_rate=0):
         # translate inversed range 1:95 to 2:32
         translated_quality = 96 - compress_quality
         translated_quality = round((((translated_quality - 1) * (31 - 2)) / (95 - 1)) + 2)
         self.output = tempfile.mkdtemp(prefix='cvat-', suffix='.data')
         target_path = os.path.join(self.output, '%d.jpg')
-        output_opts = '-start_number 0 -b:v 10000k -vsync 0 -an -y -q:v ' + str(translated_quality)
+        output_opts = '-start_number 0 -b:v 10000k -an -y -q:v ' + str(translated_quality)
+        if frame_rate > 0:
+            output_opts += ' -r ' + str(frame_rate)
+        else:
+            output_opts += ' -vsync 0'
```

Comment:
Option `-vsync 0` is very important. It allows to generate non-duplicated frames. -r can lead to duplicated frames and in general it is bad
- need to annotate the same frame twice
- there is no correspondence between frame and video. What does frame #3 mean? Is it duplication of frame #2? Is it an unique frame?

Reply:
I have tried to keep the vsync options. But if both -vsync and -r options are specified, ffmpeg reports error:
"Using -vsync 0 and -r can produce invalid output files"

Reply:
@zliang7, this is why '-r' option is not good here. See my thoughts below.